app/Listeners/Squawk/ReserveInFirProximity.php

```diff
@@ -55,12 +56,18 @@ public function handle(NetworkAircraftUpdatedEvent $event): bool
                 continue;
             }
 
-            // The aircraft is squawking its assigned code or has only recently changed code, so let it go
+            /*
+             * The aircraft is either:
+             * - Squawking its assigned code,
+             * - Squawking one of the reserved codes, so probably hasn't set its code yet
+             * - Only recently changed code, so let it go
+             */
             if (
                 (($currentAssignment = $this->squawkService->getAssignedSquawk(
                     $aircraft->callsign
                 )) && $aircraft->transponder == $currentAssignment->getCode()) ||
-                $aircraft->transponder_last_updated_at > Carbon::now()->subMinutes(self::CONSISTENT_SQUAWK_MINUTES)
+                $aircraft->transponder_last_updated_at > Carbon::now()->subMinutes(self::CONSISTENT_SQUAWK_MINUTES) ||
```

Comment:
I would be tempted to update to something like below for readability but up to you! Fits in well with the comment imo
```
$squawkingAssignedCode = (($currentAssignment = $this->squawkService->getAssignedSquawk(
                    $aircraft->callsign
                )) && $aircraft->transponder == $currentAssignment->getCode());

$squawkingNonAssignableCode = NonAssignableSquawkCode::where('code', $aircraft->transponder)->exists()

$recentlyChangedCode = $aircraft->transponder_last_updated_at > Carbon::now()->subMinutes(self::CONSISTENT_SQUAWK_MINUTES)

if ($squawkingAssignedCode || $squawkingNonAssignableCode ....
```

Reply:
Have refactored into clearer methods :)